src/consent-manager.tsx

```diff
@@ -46,27 +52,34 @@ export const injectConsentManagerApp = async (
       appContainer ??= createHTMLElement('div');
       shadowRoot.appendChild(appContainer);
 
-      // Don't inherit global styles
-      const style = createHTMLElement<HTMLStyleElement>('style');
+      if (ALLOW_INLINE_CSS) {
+        // Don't inherit global styles
+        const style = createHTMLElement<HTMLStyleElement | HTMLLinkElement>(
+          EXTERNALIZE_INLINE_CSS ? 'link' : 'style'
+        );
 
-      if (CSP_NONCE) {
-        style.nonce = CSP_NONCE;
-      }
+        if (CSP_NONCE) {
+          style.nonce = CSP_NONCE;
+        }
 
-      appContainer.appendChild(style);
+        if (EXTERNALIZE_INLINE_CSS) {
+          (style as HTMLLinkElement).rel = 'stylesheet';
+          (style as HTMLLinkElement).href = `data:text/css,${encodeURIComponent(CSS_RESET)}`;
+        }
 
-      // Append UI container to doc to activate style.sheet
-      (document.documentElement || document).append(consentManager);
+        appContainer.appendChild(style);
+        attachToDoc();
```

Comment:
had to move stuff around to keep this optimal. we want to activate the stylesheet _after_ setup for the external one, and activate it _before_ setup for the inline one

Reply:
(hence the two orthogonal ifs with a between step instead of an if else)

Reply:
just wanted to ask this, makes sense